### DISCUSSION TITLE
on.push.branch: '**' filter required to make on.push.paths filters work?

### DISCUSSION THREAD

**Discussion by papadeltasierra** [community/community#63314] [Actions]
### Select Topic Area

General

### Body

I've found some posts that imply that in order to make a workflow work if...
- one or more path file changes happen OR
- specific tags are pushed

... I also have to add a branches filter like this:
```yaml
on:
  push:
    paths:
      - 'this/dir/*'
      - 'a/tree/**'
    branches:
      - '**'
    tags:
      - 'v1.0.*'
```
Can someone please explain why the `on.push.branches` filter is required?

**Comment by ayatakinci** [comment]
The on.push.branches filter in a GitHub Actions workflow specifies the branches for which the workflow should be triggered when a push event occurs. In this case, where you want the workflow to be triggered if either path file changes occur or specific tags are pushed, the branches filter might seem redundant or confusing. lets talk about why it's used:
By default, a GitHub Actions workflow triggers only for the default branch (main or master). If you want the workflow to be triggered for changes on branches other than the default one, you need to explicitly specify the branches using the branches filter.
In your scenario, you've mentioned that you want the workflow to be triggered if either of the following conditions is met:
Path file changes happen.
Specific tags are pushed.
While paths and tags filters handle these conditions, the absence of a branches filter would mean that the workflow would only trigger for changes on the default branch (e.g., main). Adding the branches filter with a wildcard (**) ensures that the workflow triggers for changes on any branch, including both the default branch and other branches.
also including the branches filter allows your workflow to be more flexible. For example, if you decide to apply the same workflow to different branches or if you add additional conditions for workflow triggers in the future, the branches filter provides a way to control the workflow's behavior across different branches.

In summary, the on.push.branches filter is not necessarily required for every workflow, but it becomes relevant when you want your workflow to trigger on changes to branches other than the default one.

**Comment by diegocatao** [comment]
Hi @papadeltasierra,
The inclusion of the branches filter within the on.push event configuration is not always necessary, but it can serve a specific purpose in certain scenarios. The main point to understand is that the branches filter helps define the conditions under which the workflow will trigger when a push event occurs.

When you set up a workflow using the on.push event, which means the workflow will be triggered whenever there is a push to the repository. The paths filter further refines this by specifying the file paths that, when changed, should trigger the workflow.

**Comment by papadeltasierra** [comment]
Thanks both - they key point I had not spotted was
> By default, a GitHub Actions workflow triggers only for the default branch (main or master).

That explains what I was seeing and why I need the
```
branches:
- '**'
```
section.
